orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java

```diff
@@ -129,34 +133,100 @@ void testCompletion() {
   }
 
   @Test
-  void testGrounding() {
+  void testGrounding() throws IOException {
     stubFor(
-        post(anyUrl())
+        post(urlPathEqualTo("/completion"))
             .willReturn(
                 aResponse()
                     .withBodyFile("groundingResponse.json")
                     .withHeader("Content-Type", "application/json")));
-    final var response = client.chatCompletion(prompt, config);
-    final var result = response.getOriginalResponse();
-    var llmChoice =
-        ((LLMModuleResultSynchronous) result.getOrchestrationResult()).getChoices().get(0);
-
-    final var groundingData =
-        (Map<String, Object>) result.getModuleResults().getGrounding().getData();
-    assertThat(groundingData.get("grounding_query")).isEqualTo("grounding call");
-    assertThat(groundingData.get("grounding_result").toString())
-        .startsWith("Joule is the AI copilot that truly understands your business.");
-    assertThat(result.getModuleResults().getGrounding().getMessage()).isEqualTo("grounding result");
-    assertThat(((ChatMessage) result.getModuleResults().getTemplating().get(0)).getContent())
-        .startsWith(
-            "What does Joule do? Use the following information as additional context: Joule is the AI copilot that truly understands your business.");
-    assertThat(llmChoice.getMessage().getContent())
-        .startsWith(
-            "Joule is an AI copilot that revolutionizes how users interact with their SAP business systems.");
-    assertThat(llmChoice.getFinishReason()).isEqualTo("stop");
-    assertThat(llmChoice.getMessage().getContent())
-        .startsWith(
-            "Joule is an AI copilot that revolutionizes how users interact with their SAP business systems.");
+
+    final var databaseFilter =
```

Comment:
this only represents the minimum necessary config and leaves a few classes untested. While we don't need to have 100% test coverage, it would  be better to have each class at least serialized once. E.g. `GroundingFilterSearchConfiguration`, `KeyValueListPair`, `SearchDocumentKeyValueListPair`..